src/AnalyticsClientFactory.php

```diff
@@ -21,22 +22,32 @@ public static function createAuthenticatedGoogleClient(array $config): Google_Cl
     {
         $client = new Google_Client();
 
-        $client->setClassConfig(
-            'Google_Cache_File',
-            'directory',
-            $config['cache_location'] ?? storage_path('app/laravel-google-analytics/google-cache/')
-        );
+        $client->setScopes([
+            Google_Service_Analytics::ANALYTICS_READONLY,
+        ]);
 
-        $credentials = $client->loadServiceAccountJson(
-            $config['service_account_credentials_json'],
-            'https://www.googleapis.com/auth/analytics.readonly'
-        );
+        $client->setAuthConfig($config['service_account_credentials_json']);
 
-        $client->setAssertionCredentials($credentials);
+        self::configureCache($client, $config['cache']);
 
         return $client;
     }
 
+    protected static function configureCache(Google_Client $client, $config)
+    {
+        $config = collect($config);
+
+        $store = \Cache::store($config->get('store'));
+
+        $cache = new CacheItemPool($store);
+
+        $client->setCache($cache);
+
+        $client->setCacheConfig(
+            $config->except('store')->toArray()
```

Comment:
@freekmurze see this line, here I get any parameter added to the `cache` config and pass to `setCacheConfig `